Servers/WindowServer/MenuManager.cpp

```diff
@@ -120,8 +120,16 @@ void MenuManager::refresh()
 
 void MenuManager::event(Core::Event& event)
 {
-    if (WindowManager::the().active_window_is_modal())
-        return Core::Object::event(event);
+    auto* active_window = WindowManager::the().active_window();
+    if (active_window && active_window->is_modal()) {
+        if (has_open_menu()) {
+            auto* topmost_menu = m_open_menu_stack.last().ptr();
+            ASSERT(topmost_menu);
+            auto* window_menu_of = topmost_menu->window_menu_of();
+            if (window_menu_of && (active_window->window_id() != window_menu_of->window_id()))
+                return Core::Object::event(event);
```

Comment:
I think we can simplify this a little bit.
As long as the `topmost_menu` is a window menu, we can let the request through. It would be nice to put a comment here, just briefly explaining why we're doing this.

Something like this:
```cpp
// Always allow window menu interaction, even while a modal window is active.
if (!topmost_menu->window_menu_of())
    return Core::Object::event(event);
```

Reply:
@awesomekling from my reading of the code, if we rewrite it as you suggested, we'll skip handling mouse events when we have a modal window, which is what we're trying to avoid.

I did a quick check with your suggested change, and the menu doesn't receive any mouse events when a modal window is open.

I can add a comment to explain the logic, though.

Reply:
Scrap that. What you said makes sense :) I just missed the `!` while reading it.